File: tensorflow_io/image/ops/image_ops.cc
/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

#include "tensorflow/core/framework/common_shape_fns.h"
#include "tensorflow/core/framework/op.h"
#include "tensorflow/core/framework/shape_inference.h"

namespace tensorflow {

REGISTER_OP("DecodeTiffInfo")
  .Input("input: string")
  .Output("shape: int64")
  .SetShapeFn([](shape_inference::InferenceContext* c) {
    shape_inference::ShapeHandle unused;
    TF_RETURN_IF_ERROR(c->WithRank(c->input(0), 0, &unused));
    c->set_output(0, c->MakeShape({c->UnknownDim(), c->UnknownDim()}));
    return Status::OK();
  });

REGISTER_OP("DecodeTiff")
  .Input("input: string")
  .Input("index: int64")
  .Output("image: uint8")
  .SetShapeFn([](shape_inference::InferenceContext* c) {
    shape_inference::ShapeHandle unused;
    TF_RETURN_IF_ERROR(c->WithRank(c->input(0), 0, &unused));
    TF_RETURN_IF_ERROR(c->WithRank(c->input(1), 0, &unused));
    c->set_output(0, c->MakeShape({
        c->UnknownDim(), c->UnknownDim(), c->UnknownDim()}));
    return Status::OK();
  });

}  // namespace tensorflow

File: tensorflow_io/core/python/ops/image_ops.py
# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
# ==============================================================================
"""Image Ops."""

from tensorflow_io.core.python.ops import core_ops


def decode_webp(contents, name=None):
    """
    Decode a WebP-encoded image to a uint8 tensor.

    Args:
      contents: A `Tensor` of type `string`. 0-D.  The WebP-encoded image.
      name: A name for the operation (optional).

    Returns:
      A `Tensor` of type `uint8` and shape of `[height, width, 4]` (RGBA).
    """
    return core_ops.io_decode_web_p(contents, name=name)


def encode_bmp(image, name=None):
    """
    Encode a uint8 tensor to bmp image.

    Args:
      image: A Tensor. 3-D uint8 with shape [height, width, channels].
      name: A name for the operation (optional).

    Returns:
      A `Tensor` of type `string`.
    """
    return core_ops.io_encode_bmp(image, name=name)


def encode_gif(image, name=None):
    """
    Encode a uint8 tensor to gif image.

    Args:
      image: A Tensor. 3-D uint8 with shape [N, H, W, C].
      name: A name for the operation (optional).

    Returns:
      A `Tensor` of type `string`.
    """
    return core_ops.io_encode_gif(image, name=name)
